src/Dapr.Actors/Runtime/ActorManager.cs

```diff
@@ -185,18 +186,35 @@ async Task<byte[]> RequestFunc(Actor actor, CancellationToken ct)
             }
         }
 
-        internal Task FireTimerAsync(ActorId actorId, string timerName, CancellationToken cancellationToken = default)
+        internal async Task FireTimerAsync(ActorId actorId, string timerName, Stream requestBodyStream, CancellationToken cancellationToken = default)
         {
+            var timerData = await TimerInfo.DeserializeAsync(requestBodyStream);
+
+            dynamic awaitable;
             // Create a Func to be invoked by common method.
             async Task<byte[]> RequestFunc(Actor actor, CancellationToken ct)
             {
-                await
-                    actor.FireTimerAsync(timerName);
+                var actorTypeName = this.actorService.ActorTypeInfo.ActorTypeName;
+                var actorType = this.actorService.ActorTypeInfo.ImplementationType;
+                MethodInfo[] methods = actorType.GetMethods(BindingFlags.Public | BindingFlags.Instance);
+                var methodInfo = actorType.GetMethod(timerData.Callback);
+
+                var parameters = methodInfo.GetParameters();
```

Comment:
We should probably harden this code. 

- Method return type
- Number and type of parameters
- Uniqueness (no overloading allowed)